File: src/AC/1137/1137.rs
impl Solution {
    pub fn tribonacci(n: i32) -> i32 {
        let dp:Vec<i32> = vec![0, 1, 1];
        if n < 3 {
            return dp[n as usize];
        }

        let mut dp = dp;
        for i in 3..=n {
            dp.push(dp[(i - 1) as usize] + dp[(i - 2) as usize] + dp[(i - 3) as usize]);
        }

        dp[n as usize]
    }
}

File: src/AC/1137/1137.cpp
#include <vector>

class Solution {
public:
    int tribonacci(int n) {
        std::vector<int> dp(n + 1, 0);
        for (int i = 0; i <= n; i++) {
            if (i == 0)
                dp[i] = 0;
            else if (i == 1 || i == 2)
                dp[i] = 1;
            else
                dp[i] = dp[i - 1] + dp[i - 2] + dp[i - 3];
        }

        return dp[n];
    }
};